File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.cpp
class Solution {
public:
    bool isvowel(char c) {
        if (c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u' ||
            c == 'A' || c == 'E' || c == 'I' || c == 'O' || c == 'U') {
            return true;
        }
        return false;
    }

    string reverseVowels(string s) {
        int i = 0;
        int j = s.length() - 1;

        while (i < j) {
            while(i < j && !isvowel(s[i])){
                i++;
            }
            while(i < j && !isvowel(s[j])){
                j--;
            }
            if(isvowel(s[i]) && isvowel(s[j])){
                swap(s[i],s[j]);
                i++;
                j--;
            }
        }
        return s;
    }
};

File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.py
class Solution:
    def isvowel(self,c):
        if(c == 'a' or c == 'e' or c == 'i' or c == 'o' or c == 'u' 
        or c == 'A' or c == 'E' or c == 'I' or c == 'O' or c == 'U'):
            return True
        return False

    def reverseVowels(self, s: str) -> str:
        i = 0
        j = len(s) - 1
        char_list = list(s)

        while i < j:
            if not self.isvowel(char_list[i]):
                i+=1
            if not self.isvowel(char_list[j]):
                j-=1
            if self.isvowel(char_list[i]) and self.isvowel(char_list[j]):
                char_list[i], char_list[j] = char_list[j], char_list[i]
                i+=1
                j-=1
        return "".join(char_list)